// jest-joi/dist/classes/hint.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Hint = void 0;
const jest_matcher_utils_1 = require("jest-matcher-utils");
class Hint {
    text;
    options;
    constructor(matcherName, context) {
        switch (matcherName) {
            case "toMatchSchema":
                this.options = {
                    isNot: context.isNot,
                    promise: context.promise,
                };
                this.text = (0, jest_matcher_utils_1.matcherHint)(matcherName, "received", "schema", this.options);
                break;
            case "toBeSchema":
                this.options = {
                    isNot: context.isNot,
                    promise: context.promise,
                };
                this.text = (0, jest_matcher_utils_1.matcherHint)(matcherName, "received", "", this.options);
                break;
            case "toBeSchemaLike":
                this.options = {
                    isNot: context.isNot,
                    promise: context.promise,
                    comment: "Schema-like means accepted by Joi.compile()",
                };
                this.text = (0, jest_matcher_utils_1.matcherHint)(matcherName, "received", "", this.options);
                break;
        }
    }
}
exports.Hint = Hint;
