// jest-joi/dist/classes/schema.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.buildErrorMessage = exports.Schema = void 0;
const Joi = require("joi");
const utils_1 = require("../utils");
class Schema {
    input;
    compiled;
    description;
    isValid;
    error;
    constructor(schemaInput) {
        this.input = {
            value: schemaInput,
            isSimple: (0, utils_1.isSimple)(schemaInput),
            isCompiled: Joi.isSchema(schemaInput),
        };
        try {
            this.compiled = Joi.compile(this.input.value);
            this.isValid = true;
            this.error = null;
            this.description = this.compiled.describe();
        }
        catch (error) {
            this.compiled = null;
            this.error = (0, exports.buildErrorMessage)(error);
            this.isValid = false;
            this.description = {
                value: null,
                isSimple: null,
            };
        }
    }
}
exports.Schema = Schema;
const buildErrorMessage = (error) => {
    const errorString = error.toString();
    const errorContentMatch = errorString.match(/^Error: (.+)$/);
    const errorContent = errorContentMatch ? errorContentMatch[1] : errorString;
    return errorContent;
};
exports.buildErrorMessage = buildErrorMessage;
